ui/src/components/table/QTable.json

```diff
@@ -528,6 +528,30 @@
       "category": "style"
     },
 
+    "table-row-style": {
+      "type": [ "String", "Array", "Object", "Function" ],
+      "tsType": "VueStyleProp",
```

Comment:
This type would break usage into TS projects, as `VueStyleProp` doesn't accept functions
The same applies to `table-row-class` too
You should add both a custom TS types (into `types/api/qtable.d.ts`), to correctly describe the props, and params/returns section for the docs to generate correctly

IIRC `<something> | (...params) => <something>` types may generate problems, can't remember which kind of problems tho

TS API example, **untested**

```ts
export type QTableRowStyleProp = VueStyleProp | ((row: any) => VueStyleProp)
export type QTableRowClassProp = VueClassProp | ((row: any) => VueClassProp)
```

JSON API example for class, **untested**

```json
     "params": {
        "row": {
          "type": "Object",
          "desc": "The current row being processed",
          "examples": [ "{ name: 'Lorem Ipsum', price: 19 }" ]
        }
      },
      "returns": {
        "type": ["String", "Array", "Object"],
				"tsType": "VueClassProp",
        "desc": "CSS classes to apply to native HTML <tr> element",
        "examples": [ "'34f39dda-6206-4071-a9df-4393aabe49ac'", "34" ]
      },
```

Reply:
🤔 

Any ideas on how I can test this for TS?

Reply:
Create a TS project, add Volar to get template autocomplete and typechecking, then try providing a function accepting a row parameter. If it complains, the type is wrong or too strict

Reply:
Here are the props I tried:

<img width="487" alt="Screenshot 2023-01-03 at 13 53 57" src="https://user-images.githubusercontent.com/20173051/210371260-1e946f16-6599-48c1-8e3b-3d5f157a089a.png">
<img width="394" alt="Screenshot 2023-01-03 at 13 54 21" src="https://user-images.githubusercontent.com/20173051/210371367-0899db1f-17ce-44b8-ab16-978487b20adb.png">
<img width="398" alt="Screenshot 2023-01-03 at 13 54 31" src="https://user-images.githubusercontent.com/20173051/210371389-eb8515d4-0784-4604-a702-74bf95fe70a9.png">
<img width="928" alt="Screenshot 2023-01-03 at 13 54 53" src="https://user-images.githubusercontent.com/20173051/210371391-bf0bca65-dec4-484e-a091-10d6861d10e2.png">
<img width="947" alt="Screenshot 2023-01-03 at 13 55 06" src="https://user-images.githubusercontent.com/20173051/210371394-d9b197d4-ca11-4aee-83d7-c5bd539d80f7.png">


for some reason, returning null or number from a function is not a type error 😕 
<img width="571" alt="Screenshot 2023-01-03 at 15 01 59" src="https://user-images.githubusercontent.com/20173051/210372261-f5a157ff-4ca3-4de1-bd4f-86998a2c42fc.png">

Here are the updated docs:
<img width="957" alt="Screenshot 2023-01-03 at 13 58 44" src="https://user-images.githubusercontent.com/20173051/210371395-9f443975-329b-44d1-b065-836945ac50b9.png">
<img width="868" alt="Screenshot 2023-01-03 at 13 58 52" src="https://user-images.githubusercontent.com/20173051/210371399-fbf88603-77cf-485a-b66d-d6cfc3b8f4f9.png">